agent/engine/docker_client.go

```diff
@@ -125,31 +125,32 @@ type DockerClient interface {
 	ImportLocalEmptyVolumeImage() DockerContainerMetadata
 
 	// CreateContainer creates a container with the provided docker.Config, docker.HostConfig, and name. A timeout value
-	// should be provided for the request.
-	CreateContainer(*docker.Config, *docker.HostConfig, string, time.Duration) DockerContainerMetadata
+	// and a context should be provided for the request.
+	CreateContainer(*docker.Config, *docker.HostConfig, string, time.Duration, context.Context) DockerContainerMetadata
 
-	// StartContainer starts the container identified by the name provided. A timeout value should be provided for the
-	// request.
-	StartContainer(string, time.Duration) DockerContainerMetadata
+	// StartContainer starts the container identified by the name provided. A timeout value and a context should be
+	// provided for the request.
+	StartContainer(string, time.Duration, context.Context) DockerContainerMetadata
 
-	// StopContainer stops the container identified by the name provided. A timeout value should be provided for the
-	// request.
-	StopContainer(string, time.Duration) DockerContainerMetadata
+	// StopContainer stops the container identified by the name provided. A timeout value and a context should be provided
+	// for the request.
+	StopContainer(string, time.Duration, context.Context) DockerContainerMetadata
```

Comment:
here, and in all methods, please make sure that `context.Context` is the first parameter to all the methods (which is the convention in golang)

Reply:
OK